docs/src/adoc/index.adoc

```diff
@@ -277,28 +277,23 @@ or `useHandle` over `open` whenever possible.
 
 === Arguments
 
-Arguments are Jdbi's representation of SQL statement parameters, as in `select * from Foo where bar = ?`.
+Arguments are Jdbi's representation of JDBC statement parameters (the `?` in `select * from Foo where bar = ?`).
 
-For this query, you would usually bind the 1st parameter as a varchar: `'Baz'`. In Jdbi terms, this means _binding_ an _Argument_ that will insert a String into the placeholder as a varchar. Arguments can perform other bindings: a BigDecimal could be bound as a SQL decimal, a java.time.Year as a SQL int, or a complex object could be serialized to a byte array and bound as a SQL blob.
+To set a parameter `?` on a JDBC `PreparedStatement`, you would `ps.setString(1, "Baz")`.
+With Jdbi, when you bind the string `"Baz"`, it will search through all registered _ArgumentFactory_ instances
+until it finds one that is willing to convert the String into an _Argument_.
+The argument is responsible for setting the String for the placeholder exactly as `setString` does.
 
-[TIP]
-Think of Arguments as functions that map Java objects to SQL data values.
-
-In addition to classic positional parameters (`?` placeholders), Jdbi has created the concept of _named_ parameters, which can make queries with multiple parameters easier to read. Any statement can use either positional or named arguments, but they can never be mixed in the same SQL statement. In such cases, Jdbi can't be sure it's binding the right arguments, and will throw an exception.
-
-[CAUTION]
-Support for query parameters (SQL data type and location within a query) is database-specific. JDBC drivers also have inconsistent support for different Java object types. This means Arguments, while convenient, SQL-safe, and familiar in concept, are platform-specific.
```

Comment:
I moved this into the "custom arguments" section since the whole point of the argument interface is that if you aren't implementing it you don't have to deal with it